### PR TITLE
[ticket/14457] Uses a random placeholder to inject css and js

### PR DESCRIPTION
**Pull request by Nicofuma**
https://tracker.phpbb.com/browse/PHPBB3-14457

PHPBB3-14457

### CODE REVIEW

File: phpBB/phpbb/template/twig/environment.php [hunk 1]
```diff
@@ -205,26 +203,22 @@ public function render($name, array $context = [])
...
+	/**
+	 * {@inheritdoc}
+	 */
+	private function display_with_assets($name, array $context = [])
+	{
+		$placeholder_salt = unique_id();
 
-			throw $e;
-		}
+		$context['definition']->set('SCRIPTS', '__SCRIPTS_'.$placeholder_salt.'__');
```
Comment (marc1706) [inline phpBB/phpbb/template/twig/environment.php]: Spaces between $placeholder_salt, dots and quotes.

File: phpBB/phpbb/template/twig/environment.php [hunk 2]
```diff
@@ -234,10 +228,10 @@ public function display($name, array $context = [])
 	 *
 	 * @return string
 	 */
-	private function inject_assets($output)
+	private function inject_assets($output, $placeholder_salt)
 	{
-		$output = str_replace('__STYLESHEETS_PLACEHOLDER__', $this->assets_bag->get_stylesheets_content(), $output);
-		$output = str_replace('__SCRIPTS_PLACEHOLDER__', $this->assets_bag->get_scripts_content(), $output);
+		$output = str_replace('__SCRIPTS_'.$placeholder_salt.'__', $this->assets_bag->get_stylesheets_content(), $output);
```
Comment (marc1706) [inline phpBB/phpbb/template/twig/environment.php]: Missing spaces

**Comment by marc1706** [comment]
Tests need to be updated I guess.

**Comment by Nicofuma** [comment]
Updated, it was just a little mistake in my code

**Comment by marc1706** [comment]
I get this in the installer:
```
[phpBB debug] "Undefined index: rand_seed_last_update" in file /home/marc/Dokumente/phpbb/phpBB/includes/functions.php on line 107
```

**Comment by rxu** [comment]
@marc1706 Yeah, that was discussed on #phpbb-dev yesterday.
The problem is that ```unique_id()``` uses some ```$config``` values (```rand_seed```, ```rand_seed_last_update```) and tries to set them while neither values nor the database exist. Also, ```$config``` is not an object instance here (but just an array) so calling ```set()``` method on it produces error as well. For me:
```
[phpBB debug] "Undefined index: rand_seed_last_update" in file C:\Program Files\Apache\htdocs\32test1\includes\functions.php on line 107
Fatal error: Uncaught Error: Call to a member function set() on array in C:\Program Files\Apache\htdocs\32test1\includes\functions.php:109
```

**Comment by marc1706** [comment]
I randomly get this error during functional tests on sqlite3:
```
<h1>General Error</h1>                  <div>SQL ERROR [ sqlite3 ]<br /><br />library routine called out of sequence [21]<br /><br />SQL<br /><br />UPDATE phpbb_config
                        SET config_value = \'1454936713\'
                        WHERE config_name = \'rand_seed_last_update\'```

**Comment by marc1706** [comment]
Please rebase onto 3.2.x. This currently adds all the merge commits from master.

**Pull request merged**